core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -278,6 +278,15 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (pyFiles != null && !isPython) {
       SparkSubmit.printErrorAndExit("--py-files given but primary resource is not a Python script")
     }
+    if (totalExecutorCores != null && executorCores != null) {
+      val totalCores = Try(totalExecutorCores.toInt).getOrElse(-1)
```

Comment:
What is the purpose of the Try block here? if it fails the input is invalid, and proceeding with -1 can't be right.